analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/HashesShouldHaveUnpredictableSaltBase.cs

```diff
@@ -37,27 +36,30 @@ public abstract class HashesShouldHaveUnpredictableSaltBase : SymbolicRuleCheck
 
     protected override ProgramState PreProcessSimple(SymbolicContext context)
     {
-        var state = context.State;
-        var instance = context.Operation.Instance;
-        if (instance.AsObjectCreation() is { } objectCreation)
+        var state = base.PreProcessSimple(context);
+        if (context.Operation.Instance.AsObjectCreation() is { } objectCreation)
```

Comment:
This can be return and ternary now - suddenly more visible, when the noise is gone :)